plugins/mediatek-scaler/fu-mediatek-scaler-device.c

```diff
@@ -0,0 +1,375 @@
+	guint8 *buf = g_malloc0(bufsz);
+
+	if (!mediatek_scaler_device_ddc_write(self, pWrBuf, count, error))
+		return FALSE;
+
+	g_usleep(FU_MEDIATEK_SCALER_DDC_MSG_DELAY * 1000);
+	if (!fu_udev_device_pread(FU_UDEV_DEVICE(self->i2c_dev), 0x0, buf, bufsz, error)) {
+		return FALSE;
+	}
+
+	if (buf[0] == DDC_DEST_ADDR && buf[1] > DDC_DATA_LEN_DFT) {
+		gsize recvsz = buf[1] - DDC_DATA_LEN_DFT;
+		guint8 checksum = 0;
+		checksum ^= 0x50; /* slave source address */
+		for (gsize i = 0; i < recvsz + 2; i++)
+			checksum ^= buf[i];
+
+		if (checksum != buf[recvsz + 2])
+			return FALSE;
+
+		memcpy(pRdBuf, buf, recvsz);
+		return TRUE;
+	}
+
+	return FALSE;
+}
+
+static gboolean
+mediatek_scaler_display_is_connected(FuMediatekScalerDevice *self, GError **error)
+{
+	guint8 value1 = g_random_int_range(1, 255);
+	guint8 value2 = g_random_int_range(1, 255);
+	guint8 sum = value1 + value2;
+	guint8 buf[] = {0xCC, 0xFE, value1, value2};
+	g_autofree guint8 *recv = g_malloc0(10);
+
+	if (!mediatek_scaler_device_ddc_read(self, buf, sizeof(buf), recv, error))
+		return FALSE;
+
+	if (sum == recv[3])
+		return TRUE;
+
+	return FALSE;
+}
+
+static gboolean
+fu_mediatek_scaler_device_probe(FuDevice *device, GError **error)
+{
+	g_autoptr(FuUdevDevice) udev_parent = NULL;
+	g_autoptr(FuDeviceLocker) locker = NULL;
+
+	/* FuUdevDevice->probe */
+	if (!FU_DEVICE_CLASS(fu_mediatek_scaler_device_parent_class)->probe(device, error))
+		return FALSE;
+
+	/* determine the i2c_dev for dp aux dev */
+	if (!fu_mediatek_scaler_device_use_aux_dev(device, error))
+		return FALSE;
+
+	/* i2c_dev if device at interest */
+
+	/* vid and pid from PCI */
+	udev_parent = fu_udev_device_get_parent_with_subsystem(FU_UDEV_DEVICE(device), "pci");
+	if (udev_parent == NULL)
+		return FALSE;
+
+	if (!fu_device_probe(FU_DEVICE(udev_parent), error))
+		return FALSE;
+
+	fu_device_add_instance_u16(device, "VID", fu_udev_device_get_subsystem_vendor(udev_parent));
+	fu_device_add_instance_u16(device, "PID", fu_udev_device_get_subsystem_model(udev_parent));
+
+	/* add IDs */
+	fu_device_set_physical_id(device, fu_udev_device_get_device_file(FU_UDEV_DEVICE(device)));
+
+	return fu_device_build_instance_id(device, error, "I2C", "VID", "PID", NULL);
+}
+
+static gboolean
+fu_mediatek_scaler_device_probe_version(FuDevice *device, GError **error)
+{
+	FuMediatekScalerDevice *self = FU_MEDIATEK_SCALER_DEVICE(device);
+	guint32 oui;
+	guint8 version_buf[2] = {0x0};
+	g_autofree gchar *oui_string = NULL;
+	g_autofree gchar *version_str = NULL;
+	g_autoptr(FuDeviceLocker) locker = NULL;
+
+	/* open device */
+	if ((locker = fu_device_locker_new(self->i2c_dev, error)) == NULL)
+		return FALSE;
+
+	/* DPCD address 00500-00502: device OUI */
+	if (!fu_udev_device_pread(FU_UDEV_DEVICE(device), 0x500, (guint8 *)&oui, 3, error))
+		return FALSE;
+
+	oui = GUINT32_FROM_BE(oui) >> 8;
+	oui_string = g_strdup_printf("OUI:%06X", oui);
+	fu_device_add_vendor_id(device, oui_string);
```

Comment:
You should be gating the OUI on the expected one and rejecting any that aren't mediatek here

Reply:
Yeah, in a disucssion with the team if we can learn the unique ID here or elsewhere. Will update this place.

Reply:
The unique ID shouldn't be in the OUI - that should be a mediatek specific OUI.  It should be something special to indicate this stream of firmware goes with this stream of firmware already on the panel.

Reply:
Removed the reference to OUI.  I was told that the identifier for this device isn't transparent from the DPCD